[Delivers #178762750] nodejs-winrm package should support more robust shell command output and signals

brantwedel opened the pull request:
- Add onReceive with commandState and individual stream output to receive all command output (call in async loop until `commandState === 'Done'` to receive all output)
- Add ability to send signals like ctrl_c/SIGINT to commands to terminate them
- Add operationTimeout support to allow for early termination/timeouts

brantwedel committed:
Add operationTimeout param option to all commands and operations

[#178762750] nodejs-winrm package should support more robust shell command output and signals

brantwedel committed:
Add support for shell command signals like ctrl_c/SIGINT to terminate early

[#178762750] nodejs-winrm package should support more robust shell command output and signals

brantwedel committed:
Add doReceive for recieving individual stream output and state from commands

[Finishes #178762750] nodejs-winrm package should support more robust shell command output and signals

brantwedel committed:
Return exit code from powershell commands
[#178762750] nodejs-winrm package should support more robust shell command output and signals

gavinaiken commented on src/command.js:
```diff
@@ -104,23 +125,85 @@ module.exports.doReceiveOutput = async function (_params) {
+        _params.authOnce = undefined;
+    }
+
+    console.log(req);
```
is this left over from debugging?

gavinaiken reviewed (commented):
Remove that console.log if it isn't meant to be there, and :shipit:

brantwedel committed:
Remove debugging console.log, bump version number

[#178762750] nodejs-winrm package should support more robust shell command output and signals
